pkg/virt-handler/mount-manager/mount.go

```diff
@@ -0,0 +1,112 @@
+}
+
+type mountManager struct {
+	containerDiskMounter container_disk.Mounter
+	hotplugVolumeMounter hotplug_volume.VolumeMounter
+}
+
+func NewMounter(virtPrivateDir string, podIsolationDetector isolation.PodIsolationDetector, clusterConfig *virtconfig.ClusterConfig, kubeletPodsDir string) MountManager {
+	return &mountManager{
+		containerDiskMounter: container_disk.NewMounter(podIsolationDetector, clusterConfig, mountutils.NewMountRecorder(filepath.Join(virtPrivateDir, "container-disk-mount-state"))),
+		hotplugVolumeMounter: hotplug_volume.NewVolumeMounter(mountutils.NewMountRecorder(filepath.Join(virtPrivateDir, "hotplug-volume-mount-state")), kubeletPodsDir),
+	}
+}
+
+// MountsReady returns if the mount points are ready to be used
+func (m *mountManager) MountsReady(vmi *v1.VirtualMachineInstance, notInitializedSince time.Time) (bool, error) {
+	// Check container diks are ready to be mounted
+	return m.containerDiskMounter.ContainerDisksReady(vmi, notInitializedSince)
+}
+
+// Mount mounts the volumes managed directly by KubeVirt
+func (m *mountManager) Mount(vmi *v1.VirtualMachineInstance) (MountInfo, error) {
+	disksInfo, err := m.containerDiskMounter.MountAndVerify(vmi)
+	if err != nil {
+		return MountInfo{}, fmt.Errorf("failed to mount container disks: %v", err)
+	}
+
+	attachmentPodUID := types.UID("")
+	if vmi.Status.MigrationState != nil {
+		attachmentPodUID = vmi.Status.MigrationState.TargetAttachmentPodUID
+	}
+	if attachmentPodUID != types.UID("") {
+		if err = m.hotplugVolumeMounter.MountFromPod(vmi, attachmentPodUID); err != nil {
+			return MountInfo{}, fmt.Errorf("failed to mount hotplug volumes: %v", err)
+		}
+
+	} else {
+		if err = m.hotplugVolumeMounter.Mount(vmi); err != nil {
+			return MountInfo{}, fmt.Errorf("failed to mount hotplug volumes: %v", err)
+		}
+	}
+	return MountInfo{
+		containerDisksInfo: disksInfo,
+	}, nil
+}
+
+// Unmount unmounts the volumes managed directly by KubeVirt
+func (m *mountManager) Unmount(vmi *v1.VirtualMachineInstance) error {
+	errCd := m.containerDiskMounter.Unmount(vmi)
+	errHp := m.hotplugVolumeMounter.UnmountAll(vmi)
+	if errCd != nil {
+		// An error occured for both kind of volumes
+		if errHp != nil {
+			return fmt.Errorf("failed unmounting container disks: %v and hotplugged volumes: %v", errCd, errHp)
+		}
+		return errCd
+	}
+	return errHp
+}
+
+// SyncMounts mounts the volumes managed directly by KubeVirt on a running VMI
+func (m *mountManager) SyncMounts(vmi *v1.VirtualMachineInstance) error {
+	return m.hotplugVolumeMounter.Mount(vmi)
+}
+
+// SyncUnmounts unmounts the volumes managed directly by KubeVirt on a running VMI
+func (m *mountManager) SyncUnmounts(vmi *v1.VirtualMachineInstance) error {
+	return m.hotplugVolumeMounter.Unmount(vmi)
+}
+
+// IsMounted returns if a volume is mounted
+func (m *mountManager) IsMounted(vmi *v1.VirtualMachineInstance, volume string, sourceUID types.UID) (bool, error) {
```

Comment:
I think this interface method is a bit confusing. It has a generic name `IsMounted` while in fact it is applicable only for hotplug volumes. I would prefer to have a more explicit naming in such case. Perhaps `IsHotplugVolumeMounted`?

In fact, the comment applies also to `SyncMounts` and `SyncUnmounts`. IMHO, `SyncHotplugMounts` and `SyncHotplugUnmounts` would probably make things clearer.

Reply:
That's true that we only have those methods for hotplugged volumes right now. However, I was thinking that maybe in the future we'll have another mount option fitting in the syncMounts. My idea was to create this interface exactly to hide the type of volume. Please, let me know if this sounds reasonable to you.

Reply:
> My idea was to create this interface exactly to hide the type of volume.

Right, and I think in a general case it is the proper way to go. My concern here is mostly about the fact that even though container- and hotplug disks are similar, they still have some specifics which are not common. Personally, when I look at a method with a generic name such as `IsMounted`, I expect it to work for any type of volumes. Here, only after diving deeper into the implementation details, I can realize that it's for hotplug. With explicit naming, that would be clear right away.

> I was thinking that maybe in the future we'll have another mount option fitting in the syncMounts

Do you have an already shaped and specific example in mind which you could share? Because if not, then I think it's one of the [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) use-cases :)

> Please, let me know if this sounds reasonable to you.

I understand that naming is more like a matter of personal preference. Therefore, if you find the suggestion and the argumentation not strong enough, and prefer leaving it as it is now, I will not insist on changing it.

Reply:
> > I was thinking that maybe in the future we'll have another mount option fitting in the syncMounts
> 
> Do you have an already shaped and specific example in mind which you could share? Because if not, then I think it's one of the [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) use-cases :)
> 

Well, this was some kind of preparatory work for the old version of the [SCSI persistent reservation](https://github.com/kubevirt/kubevirt/pull/8210). In that case, I wanted to let virt-handler mount the socket into virt-launcher pod, but we have moved to a different solution. So, yes, this now falls into the YANI category ;)

> > Please, let me know if this sounds reasonable to you.
> 
> I understand that naming is more like a matter of personal preference. Therefore, if you find the suggestion and the argumentation not strong enough, and prefer leaving it as it is now, I will not insist on changing it.

This is an internal interface, so we can always change names of the methods. We can keep more detailed names as you suggested and always move to a more generic one if there are other mount operations to add to it. I'll update the PR accodingly

Reply:
Change the naming